### PR TITLE
fix(32941): Template tag constraints not displayed in vscode intellisense

### LINKED ISSUE #32941: Template tag constraints not displayed in vscode intellisense
```
/**
    @typedef {{
        [x:string]: any,
        testValue: number
    }} Test
*/
/**
    @type {(t: T) => number}
    @template {Test} T
*/
const getTestValue = t => t.testValue
```
Hovering over `getTestValue` displays:

const getTestValue: (test: T) => number;
_@typedef_ — Test
_@type_
{(test: T) => number}
_@template_ — T

Expected: @template T info to also contain the `{Test}` constraint

Actual: @template T info has no mention of the constraint

### PR DESCRIPTION
**Pull request by a-tarasyuk**
Fixes #32941

### CODE REVIEW

File: src/services/jsDoc.ts [hunk 1]
```diff
@@ -169,7 +169,22 @@ namespace ts.JsDoc {
...
+                const templateTag = tag as JSDocTemplateTag;
+                const displayParts: SymbolDisplayPart[] = [];
+                if (templateTag.constraint) {
+                    displayParts.push(textPart(templateTag.constraint.getText()));
+                }
+                if (length(templateTag.typeParameters)) {
+                    if (length(displayParts)) {
+                        displayParts.push(spacePart());
+                    }
+                    displayParts.push(namePart(map(templateTag.typeParameters, tp => tp.getText()).join(", ")));
```
Comment (DanielRosenwasser) [inline src/services/jsDoc.ts:181]: Won't this classify `T, U` as one `namePart`? Seems like you want to use `intersperse`

https://github.com/microsoft/TypeScript/blob/2ce05a82920aced69f31bb3ab9e7d3e710980e91/src/compiler/core.ts#L132-L146

which is already used in the same file already: https://github.com/microsoft/TypeScript/blob/aee779ac7d4eabba195e1097d05e54332e9e9d35/src/services/jsDoc.ts#L114

Comment (a-tarasyuk) [reply src/services/jsDoc.ts:181]: The `typeParameters` conversion was not changed. It is the same as it was before.
```ts
(tag as JSDocTemplateTag).typeParameters.map(tp => tp.getText()).join(", ")
```
Should `typeParameters` be returned as an array?

Comment (DanielRosenwasser) [reply src/services/jsDoc.ts:181]: I think you want

```suggestion
                    for (let i = 0, n = templateTag.typeParameters.length; i < n; i++) {
                        const tp = templateTag.typeParameters[i];
                        displayParts.push(namePart(tp.getText());
                        if (i !== n - 1) {
                            displayParts.push(punctuationPart(SyntaxKind.CommaToken));
                            displayParts.push(spacePart());
                        }
                    }
```

It's possible you should use `typeParameterNamePart`

```suggestion
                    for (let i = 0, n = templateTag.typeParameters.length; i < n; i++) {
                        const tp = templateTag.typeParameters[i];
                        displayParts.push(typeParameterNamePart(tp.getText());
                        if (i !== n - 1) {
                            displayParts.push(punctuationPart(SyntaxKind.CommaToken));
                            displayParts.push(spacePart());
                        }
                    }
```

Comment (DanielRosenwasser) [reply src/services/jsDoc.ts:181]: (it's possible that I'm missing something though - feel free to correct me if so)

Comment (a-tarasyuk) [reply src/services/jsDoc.ts]: `namePart` uses `getTagNameDisplayPart` which handles this
https://github.com/microsoft/TypeScript/blob/a9efe2be6040e581100231891dea78a7fbc4f46c/src/services/jsDoc.ts#L165

https://github.com/microsoft/TypeScript/blob/a9efe2be6040e581100231891dea78a7fbc4f46c/src/services/jsDoc.ts#L213-L214

Comment (a-tarasyuk) [reply src/services/jsDoc.ts]: @DanielRosenwasser Do I need to add additional changes?

**Review by DanielRosenwasser — APPROVED** [review]
While a `textPart` might not be as rich as getting the resolved display parts, I think this is fine for this PR.

**Review by sandersn — APPROVED** [review]

**Pull request merged**